src/features/web-compat.js

```diff
@@ -440,54 +440,30 @@ export class WebCompat extends ContentFeature {
                 return
             }
 
-            this.defineProperty(window.navigator, 'mediaSession', {
-                value: {
-                },
-                writable: true,
-                configurable: true,
-                enumerable: true
-            })
-            this.defineProperty(window.navigator.mediaSession, 'metadata', {
-                value: null,
-                writable: true,
-                configurable: false,
-                enumerable: false
-            })
-            this.defineProperty(window.navigator.mediaSession, 'playbackState', {
-                value: 'none',
-                writable: true,
-                configurable: false,
-                enumerable: false
-            })
-            this.defineProperty(window.navigator.mediaSession, 'setActionHandler', {
-                value: () => {},
-                configurable: true,
-                enumerable: true
-            })
-            this.defineProperty(window.navigator.mediaSession, 'setCameraActive', {
-                value: () => {},
-                configurable: true,
-                enumerable: true
-            })
-            this.defineProperty(window.navigator.mediaSession, 'setMicrophoneActive', {
-                value: () => {},
-                configurable: true,
-                enumerable: true
-            })
-            this.defineProperty(window.navigator.mediaSession, 'setPositionState', {
-                value: () => {},
-                configurable: true,
-                enumerable: true
+            class MyMediaSession {
+                metadata = null
+                /** @type {MediaSession['playbackState']} */
+                playbackState = 'none'
+
+                setActionHandler () {}
+                setCameraActive () {}
+                setMicrophoneActive () {}
+                setPositionState () {}
+            }
+
+            this.shimInterface('MediaSession', MyMediaSession, {
```

Comment:
Could this be just:
```
this.shimInterface(MediaSession, {...})
```
And internally it use `arg[0].name` for the typing?

Would this omit the need to then mock `.name`?

Reply:
That would require the class to always have the correct name, which we can't verify inside the method.
This file is not ready btw, the correct usages are in the test file.

Reply:
> which we can't verify inside the method.

You can with TS no?

Reply:
Just to close the loop, I am still reluctant to do this change because I fear it could be confusing. Having to name the implementation class exactly the same as the native class is not something I'd expect. This is reversible, and we can change it later.